sympy/physics/mechanics/body_base.py

```diff
@@ -0,0 +1,123 @@
+from abc import ABC, abstractmethod
+
+from sympy.core.backend import Symbol, sympify
+from sympy.physics.vector import Point
+
+__all__ = ['BodyBase']
+
+
+class BodyBase(ABC):
+    """Abstract class for body type objects."""
+    def __init__(self, name, masscenter=None, frame=None, mass=None):
+        # Note: If frame=None, no auto-generated frame is created, because a
+        # Particle does not need to have a frame by default.
+        if not isinstance(name, str):
+            raise TypeError('Supply a valid name.')
+        self._name = name
+        if mass is None:
+            mass = Symbol(f'{name}_mass')
+        if masscenter is None:
+            masscenter = Point(f'{name}_masscenter')
+        self.mass = mass
+        self.masscenter = masscenter
+        self.frame = frame
+        self.potential_energy = 0
+        self.points = []
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return (f'{self.__class__.__name__}({repr(self.name)}, masscenter='
+                f'{repr(self.masscenter)}, frame={repr(self.frame)}, mass='
+                f'{repr(self.mass)})')
```

Comment:
Non-essential, but I personally like seeing `__str__` and `__repr__` implementations at the very end of a class definition. This may not be the case in SymPy though, so if not please ignore.

Reply:
The location of magic methods in SymPy seems to be inconsistent. I've always put them at the start after the `__init__`, same seems to happen in for example `Point`, but I can also find some examples putting them at the end.